### ISSUE TITLE
Manual can bitrate.

### ISSUE THREAD

**Issue by Memfispro** [vladisenko/USB-CAN#1]
Здравствуйте! Я недавно повторил проект и мне понравился. Спасибо огромное. 
Вопрос: как мне выставит 83.33 и 33.33?
Я пробовал с калькулятором у меня не получилось

**Comment by vladisenko** [comment]
Здравствуйте.
Расчет константы для произвольного битрейта довольно мудреный получился. Он отличает от оригинального адаптера. Позже постараюсь сделать калькулятор. Для 83,3k константа равна 8A2F, а для 33,3k - AE5F.
Рекомендую обновить прошивку на последнюю - там исправлена работа виртуального последовательного порта.

**Comment by Memfispro** [comment]
Константы который вы написали очень верны и отлычно работает can adapter.
Вы не пробовали lin шину анализировать на базе slcan протокола реализовать?

**Comment by vladisenko** [comment]
Нет, еще не было необходимости анализировать LIN шину. В планах сделать новую ревизию печатной платы и добавить такую возможность.

**Comment by Memfispro** [comment]
https://github.com/uCAN-LIN/LinUSBConverter
Сделали универсально.

**Comment by Memfispro** [comment]
Скажите пожалуйста по каким формулам рассчитывается другие скорости can адаптера. С Уважением!

**Comment by vladisenko** [comment]
Я загрузил калькулятор для расчета битрейта. Попробуйте его использовать.
https://github.com/vladisenko/USB-CAN/blob/master/mask%20and%20custom%20bitrate.xlsx

**Comment by Memfispro** [comment]
Спасибо. Умно сделано однако.

**Comment by Memfispro** [comment]
Привет. На схеме я видел uart rx tx если на этот выход подключить bluetooth адаптер будет работат как bluetooth serial?

**Comment by vladisenko** [comment]
Нет, программно UART не задействован, надо править исходный код.

**Comment by Memfispro** [comment]
Понятно. Потом при тестирование адаптера не принимая пакета если отправить данные то после третьего отправки данных tx светодиод постоянно горит как будто зависает.
С Уважением.

**Comment by vladisenko** [comment]
Спасибо, действительно была ошибка. Исправил.

**Comment by Memfispro** [comment]
Быстренько проверил прошивку теперь после третьего отправки данных tx светодиод не реагирует. (Нет вспишки) После пересоединение адаптера также повторяется.

**Comment by vladisenko** [comment]
Это нормально. Если CAN шина не подключена, то и ACK некому выставлять, а следовательно, и сообщение считается не отправленным. В микроконтроллере включена повторная передача сообщения до успешной передачи. А так как всего три аппаратных mailbox-а, то четвертое сообщение уже не передается. С подключенной шиной все будет нормально отправляться.

**Comment by Memfispro** [comment]
Теперь стало понятно. Спасибо.

**Comment by iDoka** [comment]
> https://github.com/uCAN-LIN/LinUSBConverter
> Сделали универсально.

в чём универсальность?
там отдельные репо под LIN адаптер и под CAN


....а ведь могли бы!!!!!
могли бы поднять композитное USB-устройство с двумя CDC!!!
и под капотом одного STM32 были бы независимые CAN & LIN

под F0x2 не встречал композита на 2хCDC, а вот под F10х есть такой пример на гитхаб
